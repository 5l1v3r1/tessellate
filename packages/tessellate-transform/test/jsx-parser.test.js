// @flow

import fs from 'mz/fs'
import path from 'path'
import parse from '../lib/parsers/jsx-parser'

describe('jsx-parser', () => {
  it('should parse a simple JSX file', async () => {
    const buffer = await fs.readFile(path.resolve(__dirname, './fixtures/simple.jsx'))
    const result = parse(buffer.toString(), {})
    expect(result).toMatchSnapshot()
  })

  it('should parse a JSX file with custom classes', async () => {
    const buffer = await fs.readFile(path.resolve(__dirname, './fixtures/custom.jsx'))
    const result = parse(buffer.toString(), {typePrefix: 'my-prefix'})
    expect(result).toMatchSnapshot()
  })
})
