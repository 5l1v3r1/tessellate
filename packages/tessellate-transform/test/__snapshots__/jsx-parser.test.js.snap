exports[`jsx-parser should parse a JSX file with custom classes 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [],
              "props": Object {
                "alt": "Test",
                "src": "test.png",
              },
              "type": "img",
            },
          ],
          "props": Object {
            "sm": 6,
            "xs": 12,
          },
          "type": "my-prefix.Grid.Column",
        },
        Object {
          "children": Array [
            Object {
              "children": Array [],
              "props": Object {
                "args": Object {
                  "bool": true,
                  "foo": "bar",
                },
              },
              "type": "my-prefix.Custom1",
            },
            Object {
              "children": Array [],
              "props": Object {
                "args": Object {
                  "test": 4.2,
                },
              },
              "type": "my-prefix.Custom2",
            },
          ],
          "props": Object {
            "sm": 6,
            "xs": 12,
          },
          "type": "my-prefix.Grid.Column",
        },
      ],
      "props": null,
      "type": "my-prefix.Grid.Row",
    },
  ],
  "props": null,
  "type": "my-prefix.Container",
}
`;

exports[`jsx-parser should parse a simple JSX file 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        "Hello, world!",
      ],
      "props": null,
      "type": "h1",
    },
  ],
  "props": Object {
    "id": "test",
  },
  "type": "div",
}
`;
